### DISCUSSION TITLE
Undefined query param returned as a string

### DISCUSSION THREAD

**Discussion by LauraBeatris** [vercel/next.js#14813] [Help]
# Bug report

## Describe the bug

After making a request to my API route, not passing a query param, instead of returning ``undefined`` from the ``request.query`` object, it returns a **string**

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Clone https://github.com/LauraBeatris/pokedex-nextjs
2. Run ``cd server-side-rendering && yarn && yarn dev`` or ``cd server-side-rendering && npm i && npm run dev``
3. See that the pokemon list isn't returned
4. Go to the search API route to debug the result and see that ``req.query.name`` is ``"undefined"`` instead of ``undefined``

## Expected behavior

If not providing a ``req.query.name``, it should return ``undefined`` and then the filter would works as expected.

```
const filterQueryName = req.query.name
    ? new RegExp(req.query.name as string, "i")
    : /.*/;
```

## Screenshots

![Screen Shot 2020-07-03 at 06 47 22](https://user-images.githubusercontent.com/48022589/86456898-37f67a00-bcf9-11ea-8efe-91eff5589ee6.png)

## System information

- OS: [macOS]
- Version of Next.js: [e.g. 9.0.4]

**Comment by IRediTOTO** [comment]
never tried to check type of query. 
But you still can check empty string, right?

**Reply by LauraBeatris** [reply]
``/api/search`` - Will return undefined because there are no query params been passed at all  

``/api/search?name=`` - Will return an empty string 

When the user first enters the page, it won't specify any query param and then it will be undefined.

**Reply by IRediTOTO** [reply]
I think this is so clearly. You can check both of them: `if(!req.query.name||req.query.name==="")`
This will be ok

**Reply by LauraBeatris** [reply]
But it's returning ``"undefined"`` instead of ``undefined`` so ``!"undefined"`` won't work because it's not an empty string

**Reply by ParsaYadollahi** [reply]
I'm having this issue also, did you every find out the root cause of the query param returning undefined as a string instead of the undefined object itself?
